frontend/src/components/auth/components/sign-in-form/sign-in-form.tsx

```diff
@@ -1,48 +1,58 @@
-import { Button, Link } from 'components/common/common';
-import { useState } from 'react';
-import googleLogo from '../../../../assets/img/google.svg';
-import passwordEye from '../../../../assets/img/password-eye.svg';
-import '../../../../assets/css/auth.scss';
+import { Button, Input, Link, Loader, PasswordInput } from 'components/common/common';
+import googleLogo from 'assets/img/google.svg';
+import 'assets/css/auth.scss';
+import { useAppForm } from 'hooks/hooks';
+import { userSignIn } from 'validation-schemas/validation-schemas';
 
 type Props = {
-  onSubmit: () => void;
+  onSubmit: (formValues: SignInFormValues) => void;
+  isLoading: boolean;
 };
 
-type PasswordType = 'password' | 'text';
+export interface SignInFormValues {
+  email: string;
+  password: string;
+}
+
+const SignInForm: FC<Props> = ({ onSubmit, isLoading }) => {
+  const { control, errors, handleSubmit } = useAppForm<SignInFormValues>({
+    defaultValues: { email: '', password: '' },
+    validationSchema: userSignIn,
+  });
 
-const SignInForm: FC<Props> = () => {
-  const [inputPasswordType, setInputPasswordType] = useState<PasswordType>('password');
-  const handleChangeInputPasswordType = (): void => {
-    if (inputPasswordType === 'password') {
-      setInputPasswordType('text');
-    } else {
-      setInputPasswordType('password');
-    }
-  };
   return (
     <>
-      <form className="form-container">
-        <label htmlFor="sign-in-email-input">Email</label>
-        <input id="sign-in-email-input" type="email" placeholder="username@gmail.com" />
-        <label htmlFor="sign-in-password-input" className="sign-in-password-label">
-          Password
-        </label>
-        <div className="sign-in-password-container">
-          <input id="sign-in-password-input" type={inputPasswordType} placeholder="Password" />
-          <Button
-            onClick={handleChangeInputPasswordType}
-            className="check-password-btn"
-            label={<img src={passwordEye} alt="check" />}
-          />
-        </div>
+      <form className="form-container" onSubmit={handleSubmit(onSubmit)}>
+        <Input
+          control={control}
+          errors={errors}
+          name="email"
+          label="Email"
+          type="email"
+          wrapperClassName="form-input"
+          placeholder="username@gmail.com"
+        />
+        <PasswordInput
+          wrapperClassName="form-input"
+          placeholder="Password"
+          control={control}
+          name="password"
+          errors={errors}
+          label="Password"
+        />
         <Link to={AppRoute.RESTORE_PASSWORD} className="forgot-password">
           Forgot Password?
         </Link>
-        <Button className="auth-submit-btn" type="submit" label="Sign in" />
+        {isLoading ? (
+          <Loader className="submit-btn-loader" />
+        ) : (
+          <Button className="auth-submit-btn" type="submit" label="Sign in" />
+        )}
```

Comment:
Better to use a spinner on the button and change button state (loading or active)